9.js

```diff
@@ -0,0 +1,70 @@
+const findMatchingParenIndex = (string, inputIndex) => {
+  let level = 1
+  let index = inputIndex
+
+  if (string.charAt(index) === '(') {
+    index++
+
+    while(level !== 0) {
+      if (string.charAt(index) === ')') {
+        level--
+      } else if (string.charAt(index) === '(') {
+        level++
+      }
+      index++
+    }
+
+    index--
+  } else if (string.charAt(index) === ')') {
+    index--
+
+    while(level !== 0) {
+      if (string.charAt(index) === ')') {
+        level++
+      } else if (string.charAt(index) === '(') {
+        level--
+      }
+      index--
+    }
+
+    index++
+  }
+
+  return index
+}
+
+const parensBGone = string => {
```

Comment:
I recommend avoiding abbreviated identifier names - avoid making assumptions about your target developer audience.  Identifiers like these are fairly easy for a native english speaker to parse, but may be more difficult for others.  Be explicit, always.